### PR TITLE
server : fill usage info in embeddings and rerank responses

### PR DESCRIPTION
**Pull request by krystiancha**
These two commits add token usage info to responses of embedding and rerank endpoints.

### CODE REVIEW

**Review by ngxson — APPROVED** [review]
I'd appreciate if you can add a test case for it. See `test_embedding.py`

**Comment by krystiancha** [comment]
Tests added

File: examples/server/server.cpp [hunk 1]
```diff
@@ -735,14 +738,17 @@ struct server_task_result_rerank : server_task_result {
     int index = 0;
     float score = -1e6;
 
+    int32_t n_prompt_tokens;
```
Comment (ggerganov) [inline examples/server/server.cpp:741]: This seems like a better name in this context:

```suggestion
    int32_t n_tokens;
```

Same for `server_task_result_embd`.

**Review by ggerganov — APPROVED** [review]

File: examples/server/server.cpp (hunk 1 above)
Comment (krystiancha) [reply examples/server/server.cpp]: Right, `n_tokens` indeed makes more sense here.  Fixed.

**Pull request merged**